[Docs] Prepare docs for refactoring (#678)
mckinsey/vizro: vizro-ai/docs/pages/user-guides/advanced-options.md

### `return_elements`
This boolean (by default `False`) determines the return type of `vizro_ai.plot`.

If set to `False`, then dynamically generated Python code is executed to produce a `plotly.graph_objects.Figure` object from the LLM response and the user provided data frame. Strictly speaking, it produces a `vizro.charts._charts_utils._DashboardReadyFigure`, which behaves essentially like a plotly graph object, but is ready to be [inserted](add-generated-chart-usecase.md) into a [Vizro](https://vizro.readthedocs.io/en/stable/) dashboard. It also comes with the default Vizro dark theme.

If set to `True`, a pydantic model is returned from which the fig object, but also various other outputs can be generated. (see below)

### `validate_code`

> This boolean (by default `True`) determines whether the LLM generated Python code will be executed with a sample of the data in order to verify that it runs and produces a plotly figure. Be sure [to read and understand what it means when dynamically generated code is executed](../explanation/safety-in-vizro-ai.md#execution-of-dynamic-code-in-vizro-ai).

Reviewer: ```suggestion
This boolean (by default `True`) determines whether the LLM generated Python code executes with a sample of the data in order to verify that it runs and produces a plotly figure. Be sure [to read and understand what it means when dynamically generated code is executed](../explanation/safety-in-vizro-ai.md#execution-of-dynamic-code-in-vizro-ai).
```

            )
            return fig
        ```

### Obtain fig object

You can create the fig object using either of the above produced code strings (vizro or plotly), changing the chart name, and using different data. Note that when executing this function, the produced code string will be dynamically executed. Be sure [to read and understand what it means when dynamically generated code is executed](../explanation/safety-in-vizro-ai.md#execution-of-dynamic-code-in-vizro-ai).

#### Vizro ready
This fig object is in the standard `vizro_dark` theme, and can [be inserted into a Vizro dashboard](add-generated-chart-usecase.md).

!!! example "Vizro fig object"

    === "Code"
        ```py
        from vizro_ai import VizroAI
        import plotly.express as px

        df = px.data.gapminder()
        vizro_ai = VizroAI()

        res = vizro_ai.plot(df, "the trend of gdp over years in the US", return_elements=True)
        fig = res.get_fig_object(data_frame=df, vizro=True)
        fig.show()
        ```

    === "Result"
        [![VizroAIChartVizro]][VizroAIChartVizro]
        ```
    [VizroAIChartVizro]: ../../assets/user_guides/VizroAIVizro.png

#### Pure Plotly/Dash

> This fig object is a plain vanilla plotly figure.

Reviewer: ```suggestion
This fig object is a basic plotly figure.
```

Reviewer: Best to avoid colloquialisms like this

!!! example "Plotly fig object"

    === "Code"
        ```py
        from vizro_ai import VizroAI
        import plotly.express as px

        df = px.data.gapminder()
        vizro_ai = VizroAI()

        res = vizro_ai.plot(df, "the trend of gdp over years in the US", return_elements=True)
        fig = res.get_fig_object(data_frame=df, vizro=False)
        fig.show()
        ```

    === "Result"
        [![VizroAIChartPlotly]][VizroAIChartPlotly]
        ```
    [VizroAIChartPlotly]: ../../assets/user_guides/VizroAIPlotly.png

#### Using different data

> You may want create the fig object with different data. Note that of course the overall schema should still be adhered to (or else the chart might fail), but for example certain rows can be filtered out. You can re-evaluate this function to produce various fig object for different data.

Reviewer: ```suggestion
You may want create the fig object with different data. You should still follow the overall schema (or else the chart might fail) but, for example certain rows can be filtered out. You can re-evaluate this function to produce fig objects for different data.
```

!!! example "Different data"

    === "Code"
        ```py
        from vizro_ai import VizroAI
        import plotly.express as px

        df = px.data.gapminder()
        vizro_ai = VizroAI()

        res = vizro_ai.plot(df, "the average of gdp for all continents as bar chart", return_elements=True)

        # The produced chart could handle many continents, but we choose to filter for the US
        df_us = df[df['country'] == 'United States']
        fig = res.get_fig_object(chart_name="different_name", data_frame=df_us, vizro=True)
        fig.show()
        ```

    === "Result"
        [![VizroAINewData]][VizroAINewData]
        ```
    [VizroAINewData]: ../../assets/user_guides/VizroAINewData.png
        ```

#### Changing the chart name

> This option will execute the chart code with the name provided under `chart_name`. This can be important when try to avoid overwriting variables in the namespace.

Reviewer: ```suggestion
This option executes the chart code with the name provided under `chart_name`. This can be important when you want to avoid overwriting variables in the namespace.
```

### `df`
Provide any `pandas` data frame to base your query on. The LLM will receive a sample of this data frame to form an appropriate graph. 

If the option `validate_code` is set to `True` (which it is by default), the LLM created chart code will be evaluated on a sample of this data frame.

If `return_elements` is set to `False`, then the returned fig object will be created based on this (entire) data frame.

### `max_debug_retry`
This number determines how often the tool will try to correct an incorrect response (that fails various validation criteria). Under the hood this is [implemented via pydantic validators](https://docs.pydantic.dev/1.10/usage/validators/). The last response will be re-sent to the LLM together with the validation error(s) in order to receive an improved response. This concept is [inspired by the amazing instructor library](https://github.com/jxnl/instructor).

### `return_elements`
This boolean (by default `False`) determines the return type of `vizro_ai.plot`. 

> If set to `False`, then dynamically generated python code is executed in order to produce a `plotly.graph_objects.Figure` object from the LLM response and the user provided data frame. Strictly speaking, it produces a `vizro.charts._charts_utils._DashboardReadyFigure`, which behaves essentially like a plotly graph object, but is ready to be [inserted](add-generated-chart-usecase.md) into a [Vizro](https://vizro.readthedocs.io/en/stable/) dashboard. It also comes with the default Vizro dark theme.

Reviewer 2: Shall we not expose "_DashboardReadyFigure" to users? especially in how to guides

Author: You mean not mention it? Yh....

Reviewer 2: yes, `_DashboardReadyFigure ` is meant to be private no?

Author: So I thought about this for a while, I feel like we can leave it in. The reason being that this is an advanced guide, and we make it amply clear that this is only important if someone wants to know the _exact_ return type. It quickly says how it behaves in general
